config.cc

```diff
@@ -52,6 +58,14 @@ namespace
       result = QDir::fromNativeSeparators( QString::fromWCharArray( _wgetenv( L"APPDATA" ) ) );
     #else
       char const * pathInHome = ".goldendict";
+      #if QT_VERSION >= QT_VERSION_CHECK( 5, 0, 0 ) && defined( Q_OS_LINUX )
```

Comment:
Replace `Q_OS_LINUX` with `HAVE_X11` (GoldenDict's custom define) here and in other places. I think your changes are useful on BSD too.